Cleanup temperature code

thinkyhead opened the pull request:
- Get rid of unused temp states in the ISR, resulting in more frequent
  temperature reading with fewer sensors
- Shrink code slightly in min/max testing
- Include #1680 - Don't test the MAX6675 temperature if not in use
- Fix length of `raw_temp_value` array with `TEMP_SENSOR_1_AS_REDUNDANT`

thinkyhead committed:
Cleanup temperature code

- Get rid of unused temp states in the ISR, resulting in more frequent
temperature reading with fewer sensors
- Shrink code slightly in min/max testing

tonnico commented:
We should test this. When I start with the port I increased the ISR-speed and run into trouble because the temp-reading was to fast for manage_heater().

AnHardt commented:
@thinkyhead 
Please wait with this until we know why #1660 fails.
Getting wrong temperature readings even for thermistors since then.

Will have a look, into that now.

(Back again)

thinkyhead commented:
I look forward to having a proper _facepalm_ when I discover where I screwed up.

emartinez167 commented:
This explains why I have been having temperature runaway shutdowns! And I
was blaming my thermistors...
On Mon, 23 Mar 2015 at 15:58 Scott Lahteine notifications@github.com
wrote:

> I look forward to having a proper _facepalm_ when I discover where I
> screwed up.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MarlinFirmware/Marlin/pull/1674#issuecomment-84871232
> .

thinkyhead commented:
@AnHardt I'm looking through recent commits. This one was from 9 days ago. I will see if I can find others that could be implicated. https://github.com/MarlinFirmware/Marlin/commit/901c3c5392fcc1779dce53fc91cf87c675f641c7

Perhaps it would be easiest to be methodical and narrow down the specific commit that brought about this change in behavior. If it is timing-related it would be hard to pinpoint specific lines of code. But I presume the ISR runs at some known interval...?

tonnico commented:
@thinkyhead Yes, when I'm right at 128/62500 ~ 2ms. [temperature.cpp](https://github.com/MarlinFirmware/Marlin/blob/Development/Marlin/temperature.cpp#L991)

thinkyhead commented:
That bare `128` is begging to be made into a #define. But I suppose that is something we are loath to change, because so much depends on it. Or does it?

tonnico commented on Marlin/temperature.cpp:
```diff
@@ -1473,78 +1483,124 @@ ISR(TIMER0_COMPB_vect) {
-      temp_count++;
-      break;
+        temp_state = PrepareTemp_0;
+        temp_count++;
```
if not HAS_FILAMENT_SENSOR the temp_count doesn't increase.
and the temp_state doesn't reset.

thinkyhead committed:
Revert measurement loop

tonnico commented:
This Interrupt is for all buttons on the display and also for measuring the temperature. That's it. Normally no one needs to adjust this.

thinkyhead committed:
Merge branch 'Development' into fix_temperature_minmax
Latest upstream commits

thinkyhead committed:
Don't test raw temp for MAX6675

- Remove test of raw temp for thermistor 0 when using MAX6675
- Longer `raw_temp_value` array for `TEMP_SENSOR_1_AS_REDUNDANT`
